frontend/src/routes/_oh.app.tsx

```diff
@@ -156,21 +157,32 @@ function App() {
     );
   };
 
+  const doSendInitialQuery = React.useRef<boolean>(true);
+
   const sendInitialQuery = (query: string, base64Files: string[]) => {
     const timestamp = new Date().toISOString();
     send(createChatMessage(query, base64Files, timestamp));
   };
 
-  const handleOpen = React.useCallback(() => {
-    const initEvent = {
-      action: ActionType.INIT,
-      args: settings,
-    };
-    send(JSON.stringify(initEvent));
-
-    // display query in UI, but don't send it to the server
-    if (q) addIntialQueryToChat(q, files);
-  }, [settings]);
+  const handleOpen = React.useCallback(
+    (event: Event, isNewSession: boolean) => {
+      if (!isNewSession) {
+        dispatch(clearMessages());
+        dispatch(clearTerminal());
+        dispatch(clearJupyter());
+      }
```

Comment:
Wouldn't we want to clear content when starting a new session?

Reply:
When I wrote this (Yesterday), the existing workflow was that the initial query from the splash page had been added to the chat, so clearing the content at this point would also clear that query.

Reply:
Worth noting that we actually have a new dispatch courtesy of OpenHands
https://github.com/All-Hands-AI/OpenHands/blob/27716171bf56edf2f539b5da4becf0cec066937b/frontend/src/routes/_oh.app.tsx#L257-L260